Images as nodes

xevenheaven opened the pull request:
This can close #10 once it's reviewed.

xevenheaven committed:
First pass to rendering images

- Images are rendered below the nodes as of now
- TODO: Use images as nodes

xevenheaven committed:
Images implemented

- Used append g, then append image
- Node size dependent on root/middle/leaf
- TODO: Replace the empty.png

xevenheaven committed:
Fix the issue of duplicating nodes

xevenheaven committed:
Fix (& cleanup code) for node transition when kudos is created

xevenheaven committed:
Add some docs and clean everything up

xevenheaven committed:
Clean up graph.css

spinningarrow commented:
It looks pretty good on the whole. But I think the images are too big - what do you say to making them smaller?

Also, do you want to add the missing images (the team ones and the `empty`) as part of this pull request?

spinningarrow commented on src/components/TeamHub.js:
```diff
@@ -160,152 +163,110 @@ module.exports = React.createClass({
+				}
+			})
+			.attr('x', (d) => {
+				if (d.isRoot) return -r*3;
```
Since `x`, `y`, `width`, and `height` are related, can we put them in some kind of a `SIZE` constant somewhere? Probably with a different one for the root, team nodes and individual people (leaf) nodes. It'll be easier to manipulate that way.

spinningarrow replied:
Also, it might look cleaner to set the `attr`s using the object literal syntax (if http://stackoverflow.com/a/20822552/342886 is to be believed) (since the if conditions are repeating, maybe they can be refactored into a function or something? I'm just thinking out loud).

This isn't super important though, so it's up to you.

xevenheaven replied:
Agree with the constants thing. And will fiddle with the second point, the `attr`s and functions. :+1:

xevenheaven commented:
Haha your mac screen is too small. :tongue: Nah I'm kidding. Fine with it
being smaller.

Let's separate that issue with this pull request. (Not too sure when I'll
be done with the images.) Besides, still have to edit the code to use the
correct images (and perhaps remove colorbrewer).

What d'you say?
On 30 Mar 2015 21:27, "Sahil Bajaj" notifications@github.com wrote:

> It looks pretty good on the whole. But I think the images are too big -
> what do you say to making them smaller?
> 
> Also, do you want to add the missing images (the team ones and the empty)
> as part of this pull request?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/spinningarrow/kudos/pull/25#issuecomment-87680326.

spinningarrow commented:
Haha 😛That's a good point though - at some point we should make it responsive so the size of the nodes grows and shrinks with the window size perhaps (not for the first release thing though).

Yeah, I don't mind. The only thing is that then it'll be sort of broken on master but I guess that's okay?

spinningarrow commented:
Also did you change the font? It looks a bit grainy.

spinningarrow commented:
Also about the size of the images - it was more about the fact that they're all overlapping a lot! 👅

xevenheaven commented:
Yeah about it, didn't we have an issue about making the graph responsive?

The font is changed - to what, I forgot lol. Windows doesn't have as nice fonts as yours so I changed it. But it wasn't grainy when I last saw it, hmmm. But that can be changed again, np.

Technically, it isn't really _broken_ since there are still images as the parent nodes.

spinningarrow commented:
Yep, we do. I added some kind of milestones to the issues - `1` is for stuff in the first release, `2` is for things to be done later haha.

I didn't put the fonts originally, did I? I don't think they were Mac-specific. The current one is a bit bolder, but it doesn't look like it got antialiased properly. This is not a big deal though.

Haha I know I know, but you know what I mean :tongue:

xevenheaven commented:
You know what, forget what I said. I'm adding the missing images (issue #26) as part of this pull request. Hope you don't mind. Lol.

spinningarrow commented:
Haha I don't mind at all. Cheers 😁

xevenheaven committed:
Remove ugly font, add svg text-rendering

xevenheaven committed:
Use generic function to set sizes (x, y, w, h, offset)

- Also removed all usage/files related to colorbrewer

xevenheaven commented:
I've done up the images (available on Box). And implemented them in the latest push; have also removed colorbrewer. So when this PR is done, both #25 and #26 can be closed.

spinningarrow commented:
Nice work on the circles by the way! We might have to tweak some of the colours a little - currently the 'PLM' one can't be seen. No need for that to hold up this PR, though.
![screen shot 2015-04-06 at 11 36 34 pm](https://cloud.githubusercontent.com/assets/1797328/7006721/ed76661c-dcb5-11e4-834e-91fa68a9d70f.png)

spinningarrow committed:
Use const for constant values

spinningarrow committed:
Use Function.bind(this) instead of `that = this`

- Makes it easier to refactor later
